radixdlt/src/main/java/com/radixdlt/store/LedgerEntry.java

```diff
@@ -68,12 +76,13 @@ public boolean equals(Object o) {
 			return false;
 		}
 		LedgerEntry radixLedgerEntry = (LedgerEntry) o;
-		return aid.equals(radixLedgerEntry.aid);
+		return Objects.equals(aid, radixLedgerEntry.aid)
+				&& Arrays.equals(content, radixLedgerEntry.content);
 	}
 
 	@Override
 	public int hashCode() {
-		return aid.hashCode();
+		return Objects.hash(aid, Arrays.hashCode(content));
```

Comment:
This isn't as elegant as it could be.  What happens here is that `Arrays.hashCode(byte[])` calculates a primitive `int` hash, this is then boxed to an `Integer` and rehashed and combined with the hash of aid.

Would be better as something like `return Objects.hashCode(aid) ^ Arrays.hashCode(content)`, or more standardly `return Objects.hashCode(aid) * 31 + Arrays.hashCode(content)`.

In any case, it might be better to consider excluding the content from the equals and hashCode methods, as the AID should be (barring hash collisions) a representation of the content.

Reply:
removed `content`, and ignoring it in EqualsVerifier test